test/over_react/util/hoc_test.dart

```diff
@@ -27,5 +27,56 @@ main() {
 
       expect(component, isNotNull);
     });
+
+    test('reduces HOCs in the expected order', () {
+      const data0 = '0';
+      const data1 = '1';
+
+      CounterProps Function() composeHocComponent = composeHocs([
+        connect<CounterState, CounterProps>(
+          mapStateToProps: (_) => {
+            'data-0': data0,
+            'data-both': data0,
+          },
+        ),
+        connect<CounterState, CounterProps>(
+          mapStateToProps: (_) => {
+            'data-1': data1,
+            'data-both': data1,
+          },
+        )
+      ])(Counter);
+
+      CounterProps Function() verboseHoc = connect<CounterState, CounterProps>(
+        mapStateToProps: (_) => {
+          'data-0': data0,
+          'data-both': data0,
+        },
+      )(connect<CounterState, CounterProps>(
+        mapStateToProps: (_) => {
+          'data-1': data1,
+          'data-both': data1,
+        },
+      )(Counter));
+
+      final jacket = mount(
+        Dom.div()(
+          (ReduxProvider()..store = store1)(
+            (composeHocComponent()..addTestId('compose-hoc'))(),
+            (verboseHoc()..addTestId('verbose-hoc'))(),
+          ),
+        ),
+        attachedToDocument: true,
+      );
+
+      final composeHocNodeAttributes =
+          queryByTestId(jacket.getNode(), 'compose-hoc').attributes;
+      final verboseHocNode = queryByTestId(jacket.getNode(), 'verbose-hoc');
+
+      expect(composeHocNodeAttributes['data-0'], data0);
```

Comment:
#nit prefer using `containsPair`, which has nicer error messages when the expectation fails
```suggestion
      expect(composeHocNodeAttributes, containsPair('data-0', data0));
```